--- target/linux/mediatek/dts/mt7981b-cudy-tr3000-mod.dts	2026-01-10 09:11:26
+++ target/linux/mediatek/dts/mt7981b-cudy-tr3000-mod.dts	2026-01-10 09:21:39
@@ -1,20 +1,35 @@
 // SPDX-License-Identifier: GPL-2.0-or-later OR MIT
-
 /dts-v1/;
+
+/* inherit official v1 DTS */
 #include "mt7981b-cudy-tr3000-v1.dts"
 
-/ {
-	model = "Cudy TR3000 (U-Boot mod)";
-	compatible = "cudy,tr3000-mod", "mediatek,mt7981";
+/* override model + compatible */
+&{/} {
+    model = "Cudy TR3000 (U-Boot mod)";
+    compatible = "cudy,tr3000-mod", "mediatek,mt7981";
 };
 
+/* override UBI partition */
 &spi0 {
-	flash@0 {
-		partitions {
-			partition@5c0000 {
-				label = "ubi";
-				reg = <0x5c0000 0x7000000>;
-			};
-		};
-	};
+    flash@0 {
+        partitions {
+            partition@500000 {
+                label = "ubi";
+                reg = <0x500000 0x7000000>;
+            };
+        };
+    };
 };
