--- a/target/linux/mediatek/image/filogic.mk
+++ b/target/linux/mediatek/image/filogic.mk
@@ -245,6 +245,22 @@ define Device/cudy_tr3000-v1
 endef
 TARGET_DEVICES += cudy_tr3000-v1

+define Device/cudy_tr3000-v1-ubootmod
+  DEVICE_VENDOR := Cudy
+  DEVICE_MODEL := TR3000
+  DEVICE_VARIANT := v1 (ubootmod)
+  DEVICE_DTS := mt7981-cudy_tr3000-v1-ubootmod
+  DEVICE_DTS_DIR := $(DTS_DIR)/mediatek
+  DEVICE_PACKAGES := kmod-usb3 kmod-usb2 kmod-mt7981-firmware \
+	kmod-leds-gpio kmod-gpio-button-hotplug
+  IMAGE_SIZE := 32768k
+  UBINIZE_OPTS := -E 5
+  IMAGES += sysupgrade.bin
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
+endef
+TARGET_DEVICES += cudy_tr3000-v1-ubootmod


--- /dev/null
+++ b/target/linux/mediatek/dts/mt7981-cudy_tr3000-v1-ubootmod.dts
@@ -0,0 +1,12 @@
+/dts-v1/;
+
+#include "mt7981-cudy_tr3000-v1.dts"
+
+/ {
+	model = "Cudy TR3000 v1 (ubootmod)";
+	compatible = "cudy,tr3000-v1-ubootmod", "mediatek,mt7981";
+};


--- a/target/linux/mediatek/filogic/profiles.json
+++ b/target/linux/mediatek/filogic/profiles.json
@@ -120,6 +120,17 @@
       "supported_devices": ["cudy,tr3000-v1"]
     },
 
+    {
+      "id": "cudy_tr3000-v1-ubootmod",
+      "name": "Cudy TR3000 v1 (ubootmod)",
+      "image_prefix": "cudy_tr3000-v1-ubootmod",
+      "supported_devices": ["cudy,tr3000-v1-ubootmod"],
+      "profiles": {
+        "default": {
+          "description": "Default profile"
+        }
+      }
+    }
   ]
 }
